queries.sql

```diff
@@ -130,4 +130,89 @@ WHERE filtered.animals_owned =
       JOIN animals ON owners.id = animals.owner_id
       GROUP BY owners.name
     ) AS filtered
-);
\ No newline at end of file
+);
+
+-- Project 4
+
+SELECT vets.name, animals.name, visits.date_of_visit  
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON visits.vets_id = vets.id
+WHERE vets.name = 'William Tatcher'
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;  
+
+CREATE VIEW Stephanie_Mendez AS
+  SELECT animals.name, COUNT(animals.name) 
+  FROM animals
+  JOIN visits ON visits.animal_id = animals.id
+  JOIN vets Ve ON visits.vets_id = vets.id
+  WHERE vets.name = 'Stephanie Mendez'
+  GROUP BY animals.name;
+
+SELECT COUNT(name) FROM Stephanie_Mendez;
+
+SELECT vets.name, species.name
+FROM vets
+LEFT JOIN specialization ON vets.id = specialization.vets_id
+LEFT JOIN species ON species.id = specialization.species_id; 
+
+SELECT animals.name
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON visits.vets_id = vets.id
+WHERE vets.name = 'Stephanie Mendez' AND visits.date_of_visit BETWEEN '1-4-2020' AND '30-8-2020';
+
+SELECT animals.name, COUNT(animals.name) as number_of_visits
+FROM animals 
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON visits.vets_id = vets.id
+GROUP BY animals.name
+ORDER BY number_of_visits DESC
+LIMIT 1;
+
+SELECT animals.name  
+FROM animals
+JOIN visits ON Visits.animal_id = animals.id
+JOIN vets ON visits.vets_id = vets.id
+WHERE vets.name = 'Maisy Smith'
+ORDER BY visits.date_of_visit
+LIMIT 1;
+
+SELECT animals.name as animal, species.name as type, animals.date_of_birth as birth_date, animals.escape_attempts, animals.neutered, 
+animals.weight_kg, owners.name as Owner, vets.name as Vet, vets.age as Vet_age, vets.date_of_graduation, visits.date_of_visit  
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON visits.vets_id = vets.id
+JOIN species ON animals.species_id = species.id
+JOIN owners ON animals.owner_id = owners.id
+ORDER BY visits.date_of_visit DESC
+LIMIT 1;
+
+
+CREATE VIEW speciality AS
+  SELECT vets.name as Vet
+  FROM vets
+  LEFT JOIN specialization ON vets.id = specialization.vets_id
+  LEFT JOIN species ON species.id = specialization.species_id;
+
+SELECT COUNT(visits.date_of_visit)
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON vets.id = visits.vets_id
+LEFT JOIN specialization ON vets.id = specialization.vets_id
+WHERE (animals.species_id != specialization.species_id OR specialization.species_id IS NULL) AND 2 != (
+  SELECT COUNT(speciality.vet)
+  FROM speciality
+  WHERE speciality.vet = vets.name 
+);
```

Comment:
- [ ] This could be better written by doing this:

```
 SELECT count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE animals.species_id NOT IN (SELECT species_id FROM specializations WHERE vet_id = vets.id)
```

Reply:
@Gambit142 Thank you for this. I will refactor to this too.